File: bootcamps/bootcamp_2020/exercise1.py
################################################################################
################################################################################
####	exercise1.py
####
####	September 2020,  baclark@msu.edu
####	guided python example for the first exercise in bootcamp 2020
####
################################################################################

import ROOT
import numpy as np
import matplotlib.pyplot as plt
import os

# load the libAraEvent.so
ROOT.gSystem.Load(os.environ.get('ARA_UTIL_INSTALL_DIR')+"/lib/libAraEvent.so")

# open a data file
file = 

# load in the event free for this file
eventTree = 

# here's a pointer to a raw event type
rawEvent = ROOT.RawAtriStationEvent()

# now, you need to SetBranchAddress for eventTree
eventTree.

# get the number of entries in this file
num_events = 

# make a empty list where we can keep the snrs
list_of_snrs = []

# loop over the events
for event in range(num_events):
	
	# do a get entry
	eventTree.

	# check if it's a cal pulser

	isCalpulser = 

	# make a conditional statement to exit the loop if it is not a cal pulser
	if(isCalpulser==0):
		continue


	# make a *useful* event out of the *raw* event, which functionally just calibrates it
	usefulEvent = 

	# now, get the waveform from channel 0
	graph = usefulEvent.getGraphFromRFChan(0)

	# now, calculate the rms and (absolute!) peak magnitude
	# you can use some ROOT/TMath functionality if you want
	# or you can write your own code to compute the rms and peak
	# because you're in python, you might consider transferring
	# the voltages into a numpy array and using that to do the math

	rms = 
	peak = 

	# add this entry to the list
	list_of_snrs.append(peak/rms)

	# bit of cleanup
	del graph
	del usefulEvent


# and do some plotting
list_of_snrs = np.asarray(list_of_snrs)
fig = plt.figure(figsize=(8,5))
ax = fig.add_subplot(111)
ax.hist(list_of_snrs, bins=np.arange(0,10,0.5))
ax.set_ylabel('SNR')
ax.set_xlabel('Number of Events')
ax.set_title('Histogram of Ch 0 SNRs')
fig.savefig('hist_of_snr_py.png', bbox_inches='tight', dpi=300)

File: bootcamps/bootcamp_2020/exercise1.cpp
////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////
////	exercise1.cpp
////
////	September 2020,  baclark@msu.edu
////	guided cpp example for the first exercise in bootcamp 2020
////
////////////////////////////////////////////////////////////////////////////////

// C/C++ Includes
#include <iostream>

//AraRoot Includes
#include "RawAtriStationEvent.h"
#include "UsefulAtriStationEvent.h"

//ROOT Includes
#include "TTree.h"
#include "TFile.h"
#include "TGraph.h"
#include "TCanvas.h"
#include "TH1D.h"
#include "TMath.h"

using namespace std;

int main(int argc, char **argv)
{

	// for output, we will make a 1D ROOT histogram
	TH1D *h = new TH1D("snrs","snrs",20,0,10);

	// open a datafile
	TFile *input_file = 


	// load the event tree
	TTree *eventTree = 

	
	// here's a pointer to a raw event type
	RawAtriStationEvent *rawEvent=0;


	// now, you need to SetBranchAddress for eventTree
	eventTree->


	// get the number of entries
	int numEntries =


	for(int event=0; event<numEntries; event++){ //loop over those entries

		// now, do a get entry
		eventTree->


		// check if it's a cal pulser
		bool isCalpulser =


		// make a conditional statement to exit the loop if it is not a cal pulser
		if(!isCalpulser) continue;
		

		// make a *useful* event out of the *raw* event, which functionally just calibrates it
		UsefulAtriStationEvent *usefulEvent = 
	

		// now, get the waveform from channel 0
		TGraph *waveform = 


		// now, calculate the rms and (absolute!) peak magnitude
		// you can use some ROOT/TMath functionality if you want
		// or you can write your own code to compute the rms and peak
		double rms = 
		double peak = 


		// add this entry to the histogram
		h->Fill(peak/rms);


		// now, we clean up
		delete waveform;
		delete realAtriEvPtr;
	}

	// and do some plotting
	TCanvas *c = new TCanvas("c","c", 1100,850);
	h->Draw("hist");
	h->SetTitle("Histogram of Ch 0 SNRs; SNR; Number of Events");
	c->SaveAs("hist_of_snr_cpp.png");

}//close the main program
